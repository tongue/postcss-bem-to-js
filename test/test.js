const postcss = require("postcss");
const path = require("path");
const fs = require("fs");
const plugin = require("../");

const fixturesPath = path.resolve(__dirname, "./fixtures");

const run = (name, opts) => {
  let resultJs;
  const sourceFile = path.join(fixturesPath, "in", `${name}.css`);
  const sourceCss = fs.readFileSync(sourceFile).toString();

  const options = opts || {};

  options.getJs = (_, js) => {
    resultJs = js;
  };

  const plugins = [plugin(options)];
  return postcss(plugins)
    .process(sourceCss, { from: sourceFile })
    .then(result => {
      expect(result.css).toEqual(sourceCss); /* do not transform input */
      expect(resultJs).toMatchSnapshot();
    });
};

it("should handle BEM syntax", () => run("bem"));
