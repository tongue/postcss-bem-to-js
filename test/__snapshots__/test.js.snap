// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle BEM syntax 1`] = `
Object {
  "MyButton": Object {
    "$": "my-button",
    "$colorPink": "my-button--color-pink",
    "$sizeSmall": "my-button--size-small",
    "icon": Object {
      "$": "my-button__icon",
      "$colorBlue": "my-button__icon--color-blue",
      "$square": "my-button__icon--square",
    },
    "infoText": Object {
      "$": "my-button__info-text",
    },
  },
  "MyButton1A": Object {
    "$": "my-button-1-a",
  },
  "MyButton@sm": Object {
    "$": "my-button@sm",
  },
}
`;
